public/js/icinga/behavior/collapsible.js

```diff
@@ -0,0 +1,216 @@
+            'expanded'
+        )
+            .on('add', this.onExternalExpansion, this)
+            .on('delete', this.onExternalCollapse, this);
+    };
+    Collapsible.prototype = new Icinga.EventListener();
+
+    /**
+     * Initializes all collapsibles. Triggered on rendering of a container.
+     *
+     * @param event  Event  The `onRender` event triggered by the rendered container
+     */
+    Collapsible.prototype.onRendered = function(event) {
+        var _this = event.data.self;
+
+        $('.collapsible:not(.can-collapse)', event.currentTarget).each(function() {
+            var $collapsible = $(this);
+
+            // Assumes that any newly rendered elements are expanded
+            if (_this.canCollapse($collapsible)) {
+                $collapsible.after($('#collapsible-control-ghost').clone().removeAttr('id'));
+                $collapsible.addClass('can-collapse');
+
+                if (! _this.state.has(_this.icinga.utils.getCSSPath($collapsible))) {
+                    _this.collapse($collapsible);
+                }
+            }
+        });
+    };
+
+    /**
+     * Updates all collapsibles.
+     *
+     * @param event  Event  The `layout-change` event triggered by window resizing or column changes
+     */
+    Collapsible.prototype.onLayoutChange = function(event) {
+        var _this = event.data.self;
+
+        $('.collapsible').each(function() {
+            var $collapsible = $(this);
+            var collapsiblePath = _this.icinga.utils.getCSSPath($collapsible);
+
+            if ($collapsible.is('.can-collapse')) {
+                if (! _this.canCollapse($collapsible)) {
+                    $collapsible.next('.collapsible-control').remove();
+                    $collapsible.removeClass('can-collapse');
+                    _this.expand($collapsible);
+                }
+            } else if (_this.canCollapse($collapsible)) {
+                // It's expanded but shouldn't
+                $collapsible.after($('#collapsible-control-ghost').clone().removeAttr('id'));
+                $collapsible.addClass('can-collapse');
+
+                if (! _this.state.has(collapsiblePath)) {
+                    _this.collapse($collapsible);
+                }
+            }
+        });
+    };
+
+    /**
+     * A collapsible got expanded in another window, try to apply this here as well
+     *
+     * @param   {Event}     event
+     * @param   {string}    collapsiblePath
+     */
+    Collapsible.prototype.onExternalExpansion = function(collapsiblePath) {
+        var $collapsible = $(collapsiblePath);
+
+        if ($collapsible.length && $collapsible.is('.can-collapse')) {
+            this.expand($collapsible);
+        }
+    };
+
+    /**
+     * A collapsible got collapsed in another window, try to apply this here as well
+     *
+     * @param   {Event}     event
+     * @param   {string}    collapsiblePath
+     */
+    Collapsible.prototype.onExternalCollapse = function(collapsiblePath) {
```

Comment:
`onExternalCollapse` sounds like the world is going down 😆 Maybe we should stick to verbs, e.g. `onCollapseExternal` and `onExpandExternal`?